File: python/largest-local-values-in-a-matrix.py
class Solution:
    def get_local_maximal(grid: List[List[int]], x, y):
        maximal = 0
        for i in range(3):
            for j in range(3):
                maximal = max(maximal, grid[x + i][y + j])
        return maximal

    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        new_grid = []
        for x in range(len(grid) - 2):
            new_grid.append([])
            for y in range(len(grid[-1]) - 2):
                maximal = Solution.get_local_maximal(grid, x, y)
                new_grid[-1].append(maximal)
        return new_grid

File: rust/largest-local-values-in-a-matrix.rs
impl Solution {
    fn get_local_maximal(grid: &Vec<Vec<i32>>, x: usize, y: usize) -> i32 {
        let mut maximal = 0;
        for i in 0..3 {
            for j in 0..3 {
                maximal = maximal.max(grid[x + i][y + j]);
            }
        }
        return maximal;
    }

    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut new_grid = Vec::new();
        let n = grid.len() - 2;
        let m = grid[0].len() - 2;
        for x in 0..n {
            let mut row = Vec::new();
            for y in 0..m {
                let maximal = Solution::get_local_maximal(&grid, x, y);
                row.push(maximal);
            }
            new_grid.push(row);
        }
        return new_grid;
    }
}
